### DISCUSSION TITLE
Can runAsUser: 0 and Privileged Mode Be Removed in eBPF Mode for falcon-node-sensor?

### DISCUSSION THREAD

**Discussion by r3motecontrol** [CrowdStrike/falcon-helm#315] [Q&A] [answered]
Hi,

We've recently switched to the eBPF mode for the sensor, and I noticed that both the initContainer and falcon-node-sensor container are currently running as [root with privileged mode enabled](https://github.com/CrowdStrike/falcon-helm/blob/main/helm-charts/falcon-sensor/templates/daemonset.yaml#L88-L147). In the eBPF mode, since the sensor no longer uses a kernel module, do the containers still need to run as root and in privileged mode?

Additionally, are there any other security contexts (e.g., allowPrivilegeEscalation, readOnlyRootFilesystem) that can be safely removed or adjusted in this mode?

Cheers

**Comment by redhatrises** [comment, marked as answer]
Hello @r3motecontrol 

I moved this to a discussion as it makes more sense as Q&A. 

To answer your question, yes. Switching from a kernel module to an eBPF program doesn't change that. For security reasons, many parts of eBPF functionality require root permissions and privileges ; otherwise, anyone could have carte blanche access to your system. So, all the permissions that you see are still required.

**Reply by r3motecontrol** [reply]
Perfect, thank you.
